### PR TITLE
 drivers: spi: spi_ll_stm32: manage slave select and long line cleanup

### PR DESCRIPTION
**Pull request by yboutreux**
Allow the user to use software slave select instead of the hardware pin, in order to free the related GPIO and avoid unwanted SS triggering on the hardware pin. The default SS is still the hardware pin.

### CODE REVIEW

**Review by tbursztyka — APPROVED** [review]

**Review by ydamigos — APPROVED** [review]

File: drivers/spi/Kconfig.stm32 [hunk 1]
```diff
@@ -25,4 +25,10 @@ config SPI_STM32_INTERRUPT
 	help
 	  Enable Interrupt support for the SPI Driver of STM32 family.
 
+config SPI_STM32_USE_HW_SS
+	bool "STM32 Hardware Slave Select support"
+	help
```
Comment (erwango) [inline drivers/spi/Kconfig.stm32]: "help" section is usually the last one

**Review by erwango — CHANGES_REQUESTED** [review]
Ntipick

File: drivers/spi/Kconfig.stm32 (hunk 1 above)
Comment (yboutreux) [reply drivers/spi/Kconfig.stm32]: Done, thank you ! :)

**Review by erwango — APPROVED** [review]
Approved.
Due to the number of impacted boards, CI will likely never pass.

**Comment by erwango** [comment]
@galak, this one won't pass get shippable pass.
But otherwise is good.

**Comment by galak** [comment]
Merging by hand since we don't feel like this will ever complete CI.

**Pull request merged**